pkg/endpoint/local_ip.go

```diff
@@ -88,14 +92,12 @@ func GetLocalIPForDestination(dst string, family k8snet.IPFamily) string {
 	return localIP
 }
 
-func isValidGlobalIPv6(ip net.IP) bool {
-	// fd00::/8 - avoid locally assigned ULA
-	if ip.To16() == nil || ip.IsLoopback() || ip.IsLinkLocalUnicast() ||
-		ip[0] == 0xfd {
+func isAcceptableIPv6(ip net.IP) bool {
+	if ip.To16() == nil || ip.IsLoopback() || ip.IsLinkLocalUnicast() || ip.String() == infraInternalIPv6 {
```

Comment:
Is `ip.String()` guaranteed to yield the same string form as `infraInternalIPv6` as there can be different ways of representing an IPv6 address as a string wrt to abbreviating? Perhaps it would be safer to do:

```
infraInternalIPv6 = net.ParseIP("fd69::2")
...

ip = ip.To16()
return ip != nil && !ip.IsLoopback() && !ip.IsLinkLocalUnicast() && !ip.Equal(infraInternalIPv6)
```

Reply:
good point , I'll update it.